xarray/core/rolling.py

```diff
@@ -702,20 +750,20 @@ def construct(
                 raise ValueError(
                     "Either window_dim or window_dim_kwargs need to be specified."
                 )
-            window_dim = {d: window_dim_kwargs[d] for d in self.dim}
+            window_dim = {d: window_dim_kwargs[str(d)] for d in self.dim}
 
-        window_dim = self._mapping_to_list(
-            window_dim, allow_default=False, allow_allsame=False
+        window_dims = self._mapping_to_list(
+            window_dim, allow_default=False, allow_allsame=False  # type: ignore[arg-type]  # ??
```

Comment:
```suggestion
            window_dim, allow_default=False, allow_allsame=False  # type: ignore[arg-type]  # https://github.com/python/mypy/issues/12506
```
Seems similar to https://github.com/python/mypy/issues/12506.